commit bc307c630c7ca3b1d9c641568ea84e69e8494de9 (refs/changes/44/1844/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-04-25T16:45:23-07:00 (2 years, 5 months ago)
    
    TOOLS-1756 dump-hourly-sdc-data.sh breaks on zone without internal_metadata

diff --git a/tools/dump-hourly-sdc-data.sh b/tools/dump-hourly-sdc-data.sh
index 9d39da4..3806e0d 100755
--- a/tools/dump-hourly-sdc-data.sh
+++ b/tools/dump-hourly-sdc-data.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -70,7 +70,7 @@ START=$(date +%s)
 echo "$0 started at $(date -u '+%Y-%m-%dT%H:%M:%S')"
 
 TIMESTAMP=$(date -u "+%s")
-mkdir -p /var/log/sdc-data
+mkdir -p $DUMPDIR
 
 echo "Purge dumps older than a week."
 for path in $(ls $DUMPDIR/*-*.json 2>/dev/null)
@@ -94,18 +94,20 @@ sanitizeVmJson='
     this.customer_metadata = undefined;
 
     // Allow certain internal_metadata keys.
-    var iAllowedKeys = {
-        "com.joyent:ipnat_owner": true
-    };
-    var iKeys = Object.keys(this.internal_metadata);
-    var iMeta = {};
-    for (var i = 0; i < iKeys.length; i++) {
-        var iKey = iKeys[i];
-        if (iAllowedKeys[iKey]) {
-            iMeta[iKey] = this.internal_metadata[iKey];
+    if (this.internal_metadata) {
+        var iAllowedKeys = {
+            "com.joyent:ipnat_owner": true
+        };
+        var iKeys = Object.keys(this.internal_metadata);
+        var iMeta = {};
+        for (var i = 0; i < iKeys.length; i++) {
+            var iKey = iKeys[i];
+            if (iAllowedKeys[iKey]) {
+                iMeta[iKey] = this.internal_metadata[iKey];
+            }
         }
-    }
-    this.internal_metadata = iMeta;'
+        this.internal_metadata = iMeta;
+    }'
 
 # Dump VMs according to VMAPI
 echo "Dump VMAPI vms"
